File: src/app/services/prediction/influencer/FootyStatsOverFTPredictionInfluencer.ts
import {
  Bet,
  BetPredictionContext,
} from '../../../types/prediction/BetPredictionContext';
import {
  BetResultInfluencer,
  BetInfluencerCalculation,
  PreCheckReturn,
  PrecheckResult,
} from '../../../types/prediction/BetResultInfluencer';

class FootyStatsOverFTPredictionInfluencer implements BetResultInfluencer {
  // eslint-disable-next-line class-methods-use-this
  public preCheck(ctx: BetPredictionContext): PreCheckReturn {
    switch (ctx.bet) {
      case Bet.OVER_ZERO_FIVE:
      case Bet.OVER_ONE_FIVE:
        break;
      default:
        return PrecheckResult.DONT_KNOW_WHAT_TO_CALCULATE_FOR_BET;
    }
    if (!ctx.match || !ctx.match['Over05 Average']) {
      return PrecheckResult.NOT_ENOUGH_INFORMATION;
    }

    return PrecheckResult.OK;
  }

  // eslint-disable-next-line class-methods-use-this
  calculateInfluence(ctx: BetPredictionContext): BetInfluencerCalculation {
    return { amount: ctx.match['Over05 Average'] };
  }
}

export default FootyStatsOverFTPredictionInfluencer;

File: footystats-tools-backend/src/main/java/de/ludwig/footystats/tools/backend/services/prediction/influencer/FootyStatsOverFTPredictionInfluencer.java
package de.ludwig.footystats.tools.backend.services.prediction.influencer;

import de.ludwig.footystats.tools.backend.services.prediction.PrecheckResult;

public class FootyStatsOverFTPredictionInfluencer implements BetResultInfluencer{
    public PrecheckResult preCheck(BetPredictionContext ctx) {
        switch (ctx.bet()) {
            case OVER_ZERO_FIVE:
            case OVER_ONE_FIVE:
                break;
            default:
                return PrecheckResult.DONT_KNOW_WHAT_TO_CALCULATE_FOR_BET;
        }
        if (ctx.match().getOver05Average() == null) {
            return PrecheckResult.NOT_ENOUGH_INFORMATION;
        }

        return PrecheckResult.OK;
    }

    // eslint-disable-next-line class-methods-use-this
    public Float calculateInfluence(BetPredictionContext ctx) {
        return ctx.match().getOver05Average();
    }

    // eslint-disable-next-line class-methods-use-this
    public String influencerName() {
        return "FootyStatsOverFTPredictionInfluencer";
    }
}
